Library/Homebrew/dev-cmd/create.rb

```diff
@@ -51,17 +54,23 @@ def create_args
       switch "--HEAD",
              description: "Indicate that <URL> points to the package's repository rather than a file."
       flag   "--set-name=",
-             description: "Explicitly set the <name> of the new formula."
+             description:  "Explicitly set the <name> of the new formula or cask.",
+             required_for: "--cask"
       flag   "--set-version=",
-             description: "Explicitly set the <version> of the new formula."
+             description: "Explicitly set the <version> of the new formula or cask."
       flag   "--set-license=",
              description: "Explicitly set the <license> of the new formula."
       flag   "--tap=",
              description: "Generate the new formula within the given tap, specified as <user>`/`<repo>."
       switch "-f", "--force",
              description: "Ignore errors for disallowed formula names and names that shadow aliases."
 
-      conflicts "--autotools", "--cmake", "--crystal", "--go", "--meson", "--node", "--perl", "--python", "--rust"
+      conflicts "--autotools", "--cmake", "--crystal", "--go", "--meson", "--node",
+                "--perl", "--python", "--ruby", "--rust", "--cask"
+      conflicts "--cask", "--HEAD"
+      conflicts "--cask", "--set-license"
+      conflicts "--cask", "--tap"
```

Comment:
Why does `--cask` conflict with `--tap`? Can't users host casks in their own taps?

Reply:
Simply because it's not implemented yet. You could use `brew cask create my/tap/cask` previously which isn't quite compatible with the way `brew create` works.

Reply:
I'm fine with this being conflicting until this is implemented 👍🏻